yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -862,21 +941,55 @@ export class PeerManager implements PeerManagerInterface {
         return;
       }
 
+      this.markAuthHandshakeSuccess(peerId);
       this.authenticatedPeerIdToValidatorAddress.set(peerIdString, sender);
       this.authenticatedValidatorAddressToPeerId.set(sender.toString(), peerId);
       this.logger.info(
         `Successfully completed auth handshake with peer ${peerId}, validator address ${sender.toString()}`,
         logData,
       );
+
+      //Authed peers are important, it is better to mark them as direct in Gossipsub
+      this.libP2PNode.services.pubsub.direct.add(peerIdString);
```

Comment:
I wasn't sure about doing this. A supernode can largely control this already can't they themselves. Let's say they set their peer count to 200. If they set their gossipsub peering degree to 200 then they will add all authenticated peers to their mesh. Alternatively they may not want to set it that high. They have the option to set to only say 100. Still much higher than normal but this would take that option away from them.